edgedb/protocol/protocol.pyx

```diff
@@ -281,7 +404,9 @@ cdef class SansIOProtocol:
                         expect_one,
                         new_cardinality == CARDINALITY_NOT_APPLICABLE,
                         in_dc, out_dc, capabilities)
-                    re_exec = True
+
+                    if in_tid != in_dc.get_tid():
```

Comment:
I thought the server will raise a `ParameterTypeMismatchError` if the input type ids mismatch?

Reply:
Ah, this is dead code left over from an earlier iteration.